Pay/PaySession.swift

```diff
@@ -27,33 +27,36 @@
 import Foundation
 import PassKit
 
-public protocol PayControllerDelegate: class {
-    func payController(_ payController: PayController, didRequestShippingRatesFor address: PayPostalAddress, provide: @escaping (PayCheckout, [PayShippingRate]) -> Void)
-    func payController(_ payController: PayController, didSelectShippingRate shippingRate: PayShippingRate, provide: @escaping  (PayCheckout) -> Void)
-    func payController(_ payController: PayController, didCompletePayment token: Data, checkout: PayCheckout, completeTransaction: @escaping (PayController.TransactionStatus) -> Void)
-    func payControllerDidFinish(_ payController: PayController)
+public protocol PaySessionDelegate: class {
+    func paySession(_ paySession: PaySession, didRequestShippingRatesFor address: PayPostalAddress, checkout: PayCheckout, provide: @escaping (PayCheckout?, [PayShippingRate]) -> Void)
+    func paySession(_ paySession: PaySession, didSelectShippingRate shippingRate: PayShippingRate, checkout: PayCheckout, provide: @escaping (PayCheckout?) -> Void)
+    func paySession(_ paySession: PaySession, didAuthorizePayment authorization: PayAuthorization, checkout: PayCheckout, completeTransaction: @escaping (PaySession.TransactionStatus) -> Void)
+    
+    func paySessionDidFinish(_ paySession: PaySession)
 }
 
-public class PayController: NSObject {
+public class PaySession: NSObject {
     
     public enum TransactionStatus {
         case success
         case failure
     }
     
-    public weak var delegate: PayControllerDelegate?
+    public weak var delegate: PaySessionDelegate?
     
     public let merchantID: String
+    public let identifier: String
     
-    public fileprivate(set) var currency:      PayCurrency?
-    public fileprivate(set) var checkout:      PayCheckout?
-    public fileprivate(set) var shippingRates: [PayShippingRate]?
+    internal fileprivate(set) var currency:      PayCurrency?
+    internal fileprivate(set) var checkout:      PayCheckout?
```

Comment:
It can't be non-optional. A session only has a `PayCheckout` (and anything associated with one) only after a call to `authorizePaymentUsing(_: currency:)`. In theory, the checkout can also be `nil` after a payment has been completed and the session is ready to be discarded.